src/main/java/org/openrewrite/java/spring/data/MigrateAuditorAwareToOptional.java

```diff
@@ -0,0 +1,180 @@
+                if (!(statement instanceof J.Return)) {
+                    return method;
+                }
+
+                return super.visitMethodDeclaration(method, ctx);
+            }
+
+
+            @Override
+            public J.Return visitReturn(J.Return return_, ExecutionContext ctx) {
+                Expression expression = return_.getExpression();
+                if (expression instanceof J.Lambda) {
+                    J.Lambda lambda = ((J.Lambda) expression);
+                    J body = lambda.getBody();
+                    if (body instanceof J.Literal) {
+                        body = JavaTemplate.builder("Optional.ofNullable(#{any()})")
+                                .contextSensitive()
+                                .imports("java.util.Optional")
+                                .build()
+                                .apply(new Cursor(getCursor(), lambda), lambda.getCoordinates().replace(), body);
+                        return return_.withExpression(lambda.withBody(body));
+                    } else {
+                        return super.visitReturn(return_, ctx);
+                    }
+                } else if (expression instanceof J.MethodInvocation) {
+                    if (isOptional.matches(((J.MethodInvocation) expression).getMethodType().getReturnType())) {
+                        return return_;
+                    }
+                    return return_.withExpression(JavaTemplate.builder("Optional.ofNullable(#{any()})")
+                            .imports("java.util.Optional")
+                            .build()
+                            .apply(new Cursor(getCursor(), expression), expression.getCoordinates().replace(), expression));
+                } else if (expression instanceof J.NewClass && isAuditorAware.matches(((J.NewClass) expression).getClazz().getType())) {
+                    implementationVisitor.setCursor(new Cursor(getCursor(), expression));
+                    return return_.withExpression(implementationVisitor.visitNewClass((J.NewClass) expression, ctx));
+                } else if (expression instanceof J.MemberReference) {
+                    J.MemberReference memberReference = (J.MemberReference) expression;
+                    JavaType.Method methodType = memberReference.getMethodType();
+                    if (methodType == null || isOptional.matches(methodType.getReturnType())) {
+                        return return_;
+                    }
+                    Expression containing = memberReference.getContaining();
+                    //TODO Question for TIM: If I use #{any()} for the method name, as getName returns a String, I get a java.lang.ClassCastException: class java.lang.String cannot be cast to class org.openrewrite.java.tree.J
+                    JavaTemplate template = JavaTemplate.builder("() -> Optional.ofNullable(#{any()}." + methodType.getName() + "())")
```

Comment:
As indicated; if we reuse MemberReferenceToMethodInvocation here I think this will be easier to tackle.